lms/static/js/learner_dashboard/views/entitlement_unenroll_view.js

```diff
@@ -0,0 +1,159 @@
+
+                errorInfoSelector: '#unenroll-entitlement-error',
+
+                closeButtonSelector: '#unenroll-entitlement-modal .close-entitlement-modal',
+
+                mainPageSelector: '#dashboard-main',
+
+                initialize: function(options) {
+                    var _this = this;
+                    this.dashboardUrl = options.dashboardUrl;
+
+                    this.$submitButton = $(this.submitButtonSelector);
+                    this.$trackInfo = $(this.trackInfoSelector);
+                    this.$refundInfo = $(this.refundInfoSelector);
+                    this.$errorInfo = $(this.errorInfoSelector);
+
+                    this.$submitButton.on('click', this.handleSubmit.bind(this));
+
+                    $(this.triggerSelector).each(function(index) {
+                        var $trigger = $(this);
+
+                        $trigger.on('click', _this.handleTrigger.bind(_this));
+
+                        if (window.accessible_modal) {
+                            window.accessible_modal(
+                                '#' + $trigger.attr('id'),
+                                _this.closeButtonSelector,
+                                '#' + _this.$el.attr('id'),
+                                _this.mainPageSelector
+                            );
+                        }
+                    });
+                },
+
+                handleTrigger: function(event) {
+                    var $trigger = $(event.target),
+                        courseName = $trigger.data('courseName'),
+                        courseNumber = $trigger.data('courseNumber'),
+                        isRefundable = $trigger.data('entitlementIsRefundable'),
+                        apiEndpoint = $trigger.data('entitlementApiEndpoint');
+
+                    this.resetModal();
+                    this.setTrackInfo(courseName, courseNumber);
+                    this.setRefundInfo(isRefundable);
+                    this.setSubmitData(apiEndpoint, isRefundable);
+
+                    if (window.edx && window.edx.dashboard && window.edx.dashboard.dropdown) {
+                        window.edx.dashboard.dropdown.toggleCourseActionsDropdownMenu(event);
+                        this.$el.css('position', 'fixed');
+                    }
+                },
+
+                handleSubmit: function(event) {
+                    var apiEndpoint = this.$submitButton.data('entitlementApiEndpoint'),
+                        isRefundable = this.$submitButton.data('entitlementIsRefundable');
+
+                    if (apiEndpoint === undefined || isRefundable === undefined) {
+                        this.setError(gettext('Error: cannot process unenrollment request.'));
+                        return;
+                    }
+
+                    $.ajax({
+                        url: apiEndpoint,
+                        method: 'DELETE',
+                        complete: this.onComplete.bind(this)
```

Comment:
Note: I will need to change this call to comply with the API changes AJ is making (We may need to send a flag along with the request indicating that the entitlement needs to be revoked).